Integrations/Securonix/Securonix.py

```diff
@@ -838,6 +932,47 @@ def add_comment_to_incident(client: Client, args: Dict) -> Tuple[str, Dict, Dict
     return f'Comment was added to the incident {incident_id} successfully.', {}, incident
 
 
+def create_incident(client: Client, args: Dict) -> Tuple[str, Dict, Dict]:
+    """Create an incident.
+
+    Args:
+        client: Client object with request.
+        args: Usually demisto.args()
+
+    Returns:
+        Outputs.
+    """
+    violation_name = str(args.get('violation_name'))
+    resource_group = str(args.get('resource_group'))
+    resource_name = str(args.get('resource_name'))
+    entity_type = str(args.get('entity_type'))
+    entity_name = str(args.get('entity_name'))
+    action_name = str(args.get('action_name'))
+    workflow = str(args.get('workflow')) if 'workflow' in args else None
+    comment = str(args.get('comment')) if 'comment' in args else None
+    criticality = str(args.get('criticality')) if 'criticality' in args else None
+
+    if 'create incident' in action_name and not workflow:
+        raise Exception(f'Creating an incident with the action: {action_name}, Supply a workflow.')
+    response = client.create_incident_request(violation_name, resource_group, resource_name, entity_type, entity_name,
+                                              action_name, workflow, comment, criticality)
+    result = response.get('result')
+    message = response.get('messages')
+    if not result or not message or (isinstance(message, list) and 'Invalid' in message[0]):
+        if isinstance(message, list):
+            message = message[0]
+        raise Exception(f'Failed to create the incident with message:\n{message}')
```

Comment:
handle case where message == [].
this is relevant for 2 of the if statement. consider changing both.